services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -364,15 +317,32 @@ private boolean isDuplicateKeyError(final Throwable throwable) {
         return updateDevicePromise.future()
                 .map(updateResult -> {
                     if (updateResult.getDocMatched() == 0) {
-                        TracingHelper.logError(span, String.format("Device [%s] not found.", deviceId));
+                        TracingHelper.logError(span, String.format("Device [%s] not found.", deviceDto.getDeviceId()));
                         return OperationResult.empty(HttpURLConnection.HTTP_NOT_FOUND);
                     } else {
                         return OperationResult.ok(
                                 HttpURLConnection.HTTP_NO_CONTENT,
-                                Id.of(deviceId),
+                                Id.of(deviceDto.getDeviceId()),
                                 Optional.empty(),
                                 Optional.of(deviceDto.getVersion()));
                     }
                 });
     }
+
+    private <T> Future<T> isMaxDevicesLimitReached(final String tenantId) {
+        final Promise<Long> findExistingNoOfDevicesPromise = Promise.promise();
+        mongoClient.count(config.getCollectionName(), new JsonObject(), findExistingNoOfDevicesPromise);
```

Comment:
this doesn't seem to account for the tenant given but seems to count the number of elements in the collection for all tenants, doesn't it?

Reply:
My bad. I fixed it now.

Reply:
did you?

Reply:
I fixed it locally. I didn't push it yet. I will include in the next commit and push it.